test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/planner/OptaplannerIntegrationTest.java

```diff
@@ -0,0 +1,129 @@
+import org.kie.server.api.model.instance.SolverInstance;
+import org.kie.server.client.KieServicesClient;
+import org.kie.server.api.model.ReleaseId;
+import org.kie.server.client.SolverServicesClient;
+
+public class OptaplannerIntegrationTest extends AbstractCloudIntegrationTest<WorkbenchWithKieServerScenario> {
+
+    private static final ReleaseId CLOUD_BALANCE_RELEASE_ID = new ReleaseId(
+            PROJECT_GROUP_ID,
+            CLOUD_BALANCE_PROJECT_SNAPSHOT_NAME,
+            CLOUD_BALANCE_PROJECT_SNAPSHOT_VERSION);
+    private static final String CLOUD_BALANCE_SOLVER_ID = "cloudsolver";
+    private static final String CLOUD_BALANCE_SOLVER_CONFIG = "cloudbalance-solver.xml";
+
+    private static final String CLASS_CLOUD_BALANCE = "org.kie.server.testing.CloudBalance";
+    private static final String CLASS_CLOUD_COMPUTER = "org.kie.server.testing.CloudComputer";
+    private static final String CLASS_CLOUD_PROCESS = "org.kie.server.testing.CloudProcess";
+    private static final String CLASS_ADD_COMPUTER_PROBLEM_FACT_CHANGE =
+            "org.kie.server.testing.AddComputerProblemFactChange";
+    private static final String CLASS_DELETE_COMPUTER_PROBLEM_FACT_CHANGE =
+            "org.kie.server.testing.DeleteComputerProblemFactChange";
+    private static final String CLASS_CLOUD_GENERATOR = "org.kie.server.testing.CloudBalancingGenerator";
+
+    @Override protected WorkbenchWithKieServerScenario createDeploymentScenario(
+            DeploymentScenarioBuilderFactory deploymentScenarioFactory) {
+        return deploymentScenarioFactory
+                .getWorkbenchWithKieServerScenarioBuilder()
+                .withExternalMavenRepo(
+                        MavenConstants.getMavenRepoUrl(),
+                        MavenConstants.getMavenRepoUser(),
+                        MavenConstants.getMavenRepoPassword())
+                .build();
+    }
+
+    @Test
+    public void testExecuteSolver() throws Exception {
+        MavenDeployer.buildAndDeployMavenProject(
+                ClassLoader.class.getResource("/kjars-sources/cloudbalance-snapshot").getFile());
+
+        KieContainer kieContainer = KieServices.Factory.get().newKieContainer(CLOUD_BALANCE_RELEASE_ID);
+
+        KieServicesClient kieServerClient = KieServerClientProvider.getKieServerClient(
+                deploymentScenario.getKieServerDeployment(),
+                extraClasses(kieContainer));
+        kieServerClient.createContainer(CONTAINER_ID, new KieContainerResource(CONTAINER_ID, CLOUD_BALANCE_RELEASE_ID));
+
+        SolverServicesClient solverClient = kieServerClient.getServicesClient(SolverServicesClient.class);
+        SolverInstance solverInstance = solverClient.createSolver(CONTAINER_ID,
+                CLOUD_BALANCE_SOLVER_ID, CLOUD_BALANCE_SOLVER_CONFIG);
+        Assertions.assertThat(solverInstance).isNotNull();
+
+        solverInstance = solverClient.getSolver(CONTAINER_ID, CLOUD_BALANCE_SOLVER_ID);
```

Comment:
Is there any specific reason for retrieving solverInstance again?

Reply:
Good point @sutaakar . I've removed this piece of code.